test/format/wts.c

```diff
@@ -99,13 +109,14 @@ handle_message(WT_EVENT_HANDLER *handler, WT_SESSION *session, const char *messa
      */
     if ((sap = session->app_private) != NULL && sap->trace != NULL) {
         testutil_check(sap->trace->log_printf(sap->trace, "%s", message));
-        return (0);
-    }
-    if (g.trace_session != NULL) {
+        if (!printf_msg)
+            return (0);
+    } else if (g.trace_session != NULL) {
```

Comment:
Do these need to be exclusive? i.e: can there be app tracing and global tracing both enabled? I'm not familiar with the functionality..

Reply:
I don't think so. The global is the backup for when the session is not configured for trace. The code previously returned immediately meaning if the first if condition was true, it never got to the second.